pkg/gofr/config/godotenv.go

```diff
@@ -64,6 +66,14 @@ func (e *EnvLoader) read(folder string) {
 	} else {
 		e.logger.Infof("Loaded config from file: %v", overrideFile)
 	}
+
+	// Reload system environment variables to ensure they override any previously loaded values
+	for _, envVar := range os.Environ() {
+		keyValue := strings.SplitN(envVar, "=", splitParts)
+		if len(keyValue) == splitParts {
+			os.Setenv(keyValue[0], keyValue[1])
```

Comment:
Env parsing is somehow as fun as parsing JSON

You can have quote

```dotenv
FOO="bar" # this is fun
```

You can  have comments

```dotenv
FOO=bar # this is fun
```

You can have commented line

```dotenv
#FOO=bar
```

I would recommend using a well tested lib for this, godotenv is already used here. I'm unsure why you are using os.Environ and try to parse them "manually"

Reply:
@ccoVeille tried using the **godotenv** library to solve the issue, but since we already used the methods `Load()` and `Overload()` once in the code, it was not able to re-load the system environment variables. Therefore, as per the requirement of the issue, I used the above implementation manually to ensure that the system environment variables are correctly reloaded and override any previously loaded values.

Reply:
You should add a valuable comment like this one in the code.

It's not clear otherwise